activesupport/lib/active_support/hash_with_indifferent_access.rb

```diff
@@ -60,13 +60,16 @@ def with_indifferent_access
       dup
     end
 
+    # An optimization, override the initialization method `nested_under_indifferent_access`
+    # who was defined in `activesupport/lib/active_support/core_ext/hash/indifferent_access.rb`.
+    # If `self.is_a?(HashWithIndifferentAccess)` is true, we don't need to instantiate again.
     def nested_under_indifferent_access
       self
     end
 
     def initialize(constructor = {})
       if constructor.respond_to?(:to_hash)
-        super()
+        super
```

Comment:
This is actually subtly different.  `super()` calls the superclass's `initialize` without any args, but `super` calls the superclass's `initialize` with all defined args to, i.e. `super(constructor)` in this case.

Reply:
Oh, that is my fault. Thanks for your guide.